foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java

```diff
@@ -0,0 +1,283 @@
+                Log.d(TAG, "shouldOverrideUrlLoading: " + url);
+                Uri uri = Uri.parse(url);
+
+                /*
+                 * Check if redirect url is of the form
+                 * foursquareauth://callback?code=CODE&error=ERROR, then extract
+                 * the code or error code from it.
+                 */
+                if (URI_SCHEME.equals(uri.getScheme())
+                        && URI_AUTHORITY.equals(uri.getAuthority())) {
+                    String code = uri.getQueryParameter("code");
+                    String error = uri.getQueryParameter("error");
+                    Intent result = new Intent();
+
+                    if (TextUtils.isEmpty(error)) {
+                        result.putExtra(FoursquareOAuth.INTENT_RESULT_CODE, code);
+
+                    } else {
+                        if (ERROR_CODE_ACCESS_DENIED.equals(error)) {
+                            result.putExtra(FoursquareOAuth.INTENT_RESULT_DENIED, true);
+                        } else {
+                            result.putExtra(FoursquareOAuth.INTENT_RESULT_ERROR, error);
+                        }
+                    }
+
+                    setResult(Activity.RESULT_OK, result);
+                    finish();
+                    return true;
+                }
+                return false;
+            }
+        });
+
+        CookieSyncManager.createInstance(this);
+        setCookies(createCookiesArray());
+        ensureCookieSyncManagerAvailable();
+        CookieSyncManager.getInstance().sync();
+
+        webView.loadUrl(String.format(OAUTH_URL, Uri.encode(clientId), Uri.encode(appSignature)));
+    }
+
+    private void onInvalidConnectRequest(String message) {
+        Log.e(TAG, message);
+
+        Intent data = new Intent();
+        data.putExtra(FoursquareOAuth.INTENT_RESULT_ERROR, ERROR_CODE_INVALID_REQUEST);
+        data.putExtra(FoursquareOAuth.INTENT_RESULT_ERROR_MESSAGE, getString(R.string.invalid_connect_request));
+        setResult(Activity.RESULT_OK, data);
+        finish();
+    }
+
+    private void onUnsupportedVersion(String clientId) {
+        Log.e(TAG, "Library version is not supported.");
+        final String referrer = String.format(MARKET_REFERRER, clientId);
+        startActivity(new Intent(Intent.ACTION_VIEW,
+                Uri.parse(URI_MARKET_PAGE)
+                        .buildUpon()
+                        .appendQueryParameter("referrer", referrer)
+                        .build()
+        ));
+
+        Intent data = new Intent();
+        data.putExtra(FoursquareOAuth.INTENT_RESULT_ERROR, ERROR_CODE_UNSUPPORTED_VERSION);
+        data.putExtra(FoursquareOAuth.INTENT_RESULT_ERROR_MESSAGE, getString(R.string.unsupported_version));
+        setResult(Activity.RESULT_OK, data);
+        finish();
+    }
+
+    /**
+     * Called when the Activity is visible to the user and actively running.
+     * Resumes the WebView.
+     */
+    @SuppressLint("NewApi")
```

Comment:
Which class has this issue, and how it the API not existing being handled?
Is it `ensureCookieSyncManagerAvailable()`?

Reply:
not sure. this is copied from foursquare-android